### PR TITLE
Feature/dependency check

### PR DESCRIPTION
**Pull request by chrisdutz**
Added a check to the build, that fails the build if the list of dependencies reported by the sbom project doesn't match the list on file in the "dependencies.json" file in the root of the project. This way PRs changing the dependencies also need to update this file, which makes it more obvious which dependencies have been added or removed from the build in a PR.

As the usage of the maven "-pl" and "-am" have a huge impact on the content the sbom plugin will report, it doesn't really make much sense to have this check active per default, so I've implemented the check to only be active by setting the property `dependencyCheck.skip` to true. So by adding `-DdependencyCheck.skip=true" to the commandline.

In order to ensure the check is run for a PR, I've added a new github-actions pipeline, which only has the goal to do this check.

Here's an example on what the build output produces with enabled check if I added one dependency (plc4j-spi) to the build without updating the dependencies.json file.

image

### CODE REVIEW

File: src/main/xslt/sbom-filter.xsl [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    xmlns:sbom="http://cyclonedx.org/schema/bom/1.4">
+    
+    <xsl:output
+        method="text"
+        indent="no"
+        encoding="utf-8"
+    />
+    
+    <xsl:template match="/">{<!--
+    "project": [<xsl:for-each select="sbom:bom/sbom:components/sbom:component[sbom:group='org.apache.iotdb']">
```
Comment (OneSizeFitsQuorum) [inline src/main/xslt/sbom-filter.xsl:33]: Shall we remove this annotated lines?

Comment (chrisdutz) [reply src/main/xslt/sbom-filter.xsl:33]: I thought I would leave it in there so it could be commented in quickly. But sure ... I'll remove it.

Comment (chrisdutz) [reply src/main/xslt/sbom-filter.xsl]: Removed

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**